### PR TITLE
Visit Opening when patient searched

### PR DESCRIPTION
**Pull request by Johngtka**
Features list:
- [x] `autocomplete can open visit when patient was searched`

ezgif com-video-to-gif

### CODE REVIEW

**Comment by Johngtka** [comment]
S&M commit proposal: 
`feat: added visit opening when patient searched (#44)`

File: src/app/patient-search/patient-search.component.ts [hunk 1]
```diff
@@ -69,14 +72,21 @@ export class PatientSearchComponent implements OnInit {
...
     }
 
     onSelected() {
         this.selectedPatient.emit(this.searchPatientsCtrl.value);
+        this.searchedPatient(this.searchPatientsCtrl.value);
+    }
+    private searchedPatient(patient: Patient) {
+        this.route.navigate(['visit'], {
+            state: patient,
+        });
```
Comment (macrog) [inline src/app/patient-search/patient-search.component.ts:88]: this doesn't make sense. on line 82 you are doing `.emit` to notify any of the `component` that this component is a child of about that new patient has been selected, yet you added a specific code that always will reroute to the `/visit` 😕 That makes this patient search component tightly couple to the `/visit` route and that is not what we wanted. Now when you look for a patient in history and expect to be shown a a selected patient's history you get rerouted to `/visit` 

ShareX_d1A09a9B5h

To complete this task you need to apply the same logic but in the component that we are interested in to reroute to `/visit`

**Review by macrog — CHANGES_REQUESTED** [review]

File: src/app/patient-search/patient-search.component.ts (hunk 1 above)
Comment (Johngtka) [reply src/app/patient-search/patient-search.component.ts:88]: explain in points

File: src/app/patient-search/patient-search.component.ts [hunk 2]
```diff
@@ -34,10 +34,10 @@ export class PatientSearchComponent implements OnInit {
     isLoading = false;
     errorMsg!: string;
     minLengthTerm = 3;
+    patient: Patient;
```
Comment (macrog) [inline src/app/patient-search/patient-search.component.ts:37]: `patient` is not used, can be removed!

**Review by macrog — CHANGES_REQUESTED** [review]

**Review by macrog — APPROVED** [review]
👍

**Pull request merged**